tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/KnowledgeGuardian.scala

```diff
@@ -17,65 +17,33 @@ import org.scalajs.core.ir.Trees.FieldDef
 import org.scalajs.core.tools.linker._
 
 private[emitter] object KnowledgeGuardian {
-  private[KnowledgeGuardian] trait Unregisterable {
+  trait Unregisterable {
     def unregister(invalidatable: Invalidatable): Unit
   }
 
   trait Invalidatable {
-    private[KnowledgeGuardian] def registeredTo(
-        unregisterable: Unregisterable): Unit
+    def registeredTo(unregisterable: Unregisterable): Unit
```

Comment:
Hmmm... Sadness... Moving it into the class also doesn't help? IMO that would be acceptable and better than having public stuff because Scalac can't handle it.